lib/device/repositories/device_location_repository.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:hnh/domain/repositories/location_repository.dart';
+import 'package:hnh/domain/entities/location.dart';
+import 'dart:async';
+import 'package:location/location.dart' as LocationLib;
+
+class DeviceLocationRepository implements LocationRepository {
+  //This class is a "singleton", it will only be instantiated once (the _instance)
+  static final DeviceLocationRepository _instance = DeviceLocationRepository._internal();
+  DeviceLocationRepository._internal();
+  factory DeviceLocationRepository() => _instance;
+
+  /// getLocation
+  /// Gets the current location of the device in the format of Location entity
+  /// Returns: A Future<Location> object or null upon failure.
+  Future<Location> getLocation() async{
+    //Instantiate the library's class
+    LocationLib.Location location_getter = new LocationLib.Location();
+    //This is the location as the library returns it, a map of string,double
+    Map location = <String, double>{};
+    //This is the format we will need the location in when we return it
+    Location result;
+    try {
+      location = await location_getter.getLocation();
+      result = new Location(location['latitude'].toString(), location['longitude'].toString(), (new DateTime.now().millisecondsSinceEpoch / 1000).toString());
+      return result;
+    }catch(e){
+      print("Fuck.");
+      print(e);
+      return null;
```

Comment:
`rethrow` to throw the error back since the usecases will expect an error to be thrown if something went wrong instead of returning `null`

Reply:
Also delete the print while we're at it

Reply:
Done.